pkg/publish/default_test.go

```diff
@@ -93,133 +74,82 @@ func md5Hash(s string) string {
-		case "/v2/":
-			w.WriteHeader(http.StatusOK)
-		case initiatePath:
-			if r.Method != http.MethodPost {
-				t.Errorf("Method; got %v, want %v", r.Method, http.MethodPost)
-			}
-			http.Error(w, "Mounted", http.StatusCreated)
-		case manifestPath:
-			if r.Method != http.MethodPut {
-				t.Errorf("Method; got %v, want %v", r.Method, http.MethodPut)
-			}
-			http.Error(w, "Created", http.StatusCreated)
-		default:
-			t.Fatalf("Unexpected path: %v", r.URL.Path)
+		tag, err := name.NewTag(fmt.Sprintf("%s/%s:latest", u.Host, expectedRepo))
+		if err != nil {
+			t.Fatalf("NewTag() = %v", err)
 		}
-	}))
-	defer server.Close()
-	u, err := url.Parse(server.URL)
-	if err != nil {
-		t.Fatalf("url.Parse(%v) = %v", server.URL, err)
-	}
-	tag, err := name.NewTag(fmt.Sprintf("%s/%s:latest", u.Host, expectedRepo))
-	if err != nil {
-		t.Fatalf("NewTag() = %v", err)
-	}
 
-	repoName := fmt.Sprintf("%s/%s", u.Host, base)
+		repoName := fmt.Sprintf("%s/%s", u.Host, base)
 
-	def, err := NewDefault(repoName, WithNamer(md5Hash))
-	if err != nil {
-		t.Errorf("NewDefault() = %v", err)
-	}
-	if d, err := def.Publish(img, build.StrictScheme+importpath); err != nil {
-		t.Errorf("Publish() = %v", err)
-	} else if !strings.HasPrefix(d.String(), repoName) {
-		t.Errorf("Publish() = %v, wanted prefix %v", d, tag.Repository)
-	} else if !strings.HasSuffix(d.Context().String(), md5Hash(strings.ToLower(importpath))) {
-		t.Errorf("Publish() = %v, wanted suffix %v", d.Context(), md5Hash(importpath))
+		def, err := NewDefault(repoName, WithNamer(md5Hash))
+		if err != nil {
+			t.Errorf("NewDefault() = %v", err)
+		}
+		if d, err := def.Publish(br, build.StrictScheme+importpath); err != nil {
+			t.Errorf("Publish() = %v", err)
+		} else if !strings.HasPrefix(d.String(), repoName) {
+			t.Errorf("Publish() = %v, wanted prefix %v", d, tag.Repository)
+		} else if !strings.HasSuffix(d.Context().String(), md5Hash(strings.ToLower(importpath))) {
+			t.Errorf("Publish() = %v, wanted suffix %v", d.Context(), md5Hash(importpath))
+		}
 	}
 }
-
 func TestDefaultWithTags(t *testing.T) {
-	img, err := random.Image(1024, 1)
-	if err != nil {
-		t.Fatalf("random.Image() = %v", err)
-	}
-	base := "blah"
-	importpath := "github.com/Google/go-containerregistry/cmd/crane"
-	expectedRepo := fmt.Sprintf("%s/%s", base, strings.ToLower(importpath))
-	headPathPrefix := fmt.Sprintf("/v2/%s/blobs/", expectedRepo)
-	initiatePath := fmt.Sprintf("/v2/%s/blobs/uploads/", expectedRepo)
-	manifestPath := fmt.Sprintf("/v2/%s/manifests/", expectedRepo)
-
-	createdTags := make(map[string]struct{})
-
-	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.Method == http.MethodHead && strings.HasPrefix(r.URL.Path, headPathPrefix) && r.URL.Path != initiatePath {
-			http.Error(w, "NotFound", http.StatusNotFound)
-			return
+	for _, br := range []build.Result{img, idx} {
+		base := "blah"
+		importpath := "github.com/Google/go-containerregistry/cmd/crane"
```

Comment:
```suggestion
		importpath := "github.com/google/go-containerregistry/cmd/crane"
```

Reply:
This was for https://github.com/google/ko/blob/bec089d9c82eadfbd3e772395b1581709bc2055a/pkg/publish/default.go#L100-L101